add saving in ArchivesSpace information

p-galligan opened the pull request:
Adding general information to the saving in ArchivesSpace section. @daremiyo and I spoke and don't see a ton else to add to this section unless we're misunderstanding the goal here. Fixes #11 

### Questions/Comments

Is this the information we're looking for here? Does this cross over with modification sections? I think the general confusion that we had about this project was where one section ends and another begins. Not sure this is the the information that others are looking for.

Commit:
add saving in ArchivesSpace information

Adding general information to the saving in ArchivesSpace section. @daremiyo and I spoke and don't see a ton else to add to this section unless we're misunderstanding the goal here.

helrond commented on README.md:
```diff
@@ -12,6 +12,12 @@
+The RAC_ASpace library improves upon the ArchivesSnake saving functionality by gerneralizing processes and making it easier to save new JSON data and push that back up to ArchivesSpace.
+
+We plan to do that by:
+  * Generalizing the transformation from a JSONModel object fetched by ASNake to editable JSON
```
Is there anything more involved than just calling the `json()` method on the `JSONModel` object?

p-galligan replied:
No, I don't really think there is, so I'm not sure how much work there is to do here. I think there is maybe potential to tie calling that method into some other actions, but that might crossover with fetching.

helrond commented on README.md:
```diff
@@ -12,6 +12,12 @@
+
+We plan to do that by:
+  * Generalizing the transformation from a JSONModel object fetched by ASNake to editable JSON
+  * Simplify the process of replacing JSON content through the use of automated methods
```
I'm curious about this. What kinds of things do you have in mind? I think this is where there might potentially be overlap with what we're calling data helpers.

p-galligan replied:
Yeah, I think there's some potential overlap here. I think what @daremiyo and I were thinking is that there's potential to move from abstraction layer to client layer and make updates in fewer steps that we're currently doing. I think there's potential to clean up the language here for clarity.

helrond reviewed (commented):
Added some comments which are not necessarily blockers, more requests for further information.

p-galligan committed:
clarify language

add clarity to saving point 2

HaSistrunk committed:
standardize language

helrond commented:
I've read 

>Simplify the process of saving new JSON content to arrays by automated processes to help create the proper data structure and by programmatically identifying the correct index location for new material

several times and I'm still unsure what this means

p-galligan commented:
Lol, @helrond , I've just re-read, and you're right, it doesn't make a ton of sense. I was trying to get to the point of saying there might be a better way to traverse and update complicated data fields within an AS object. For instance, arrays within arrays, like in multipart notes with subnotes, etc. And, possibly determining the index locations for new items added in those situations. I think it's not necessarily fully thought out, and I'd be happy to remove it, especially since it feels like maybe a crossover between data helpers/fetchers.

helrond commented:
OIC. Yeah that feels like a data helpers thing to me.

I'm also somewhat idly wondering if maybe part of what we need to look at developing is something of a pattern library for how you do things like replace note text in multipart notes.

p-galligan commented:
I think that's what I was trying to get at here. I'll remove that bullet point.

p-galligan committed:
removing content that would fit better in data helpers

Removing bullet point 2 that was unclear and would have fit better in a data helpers section.

HaSistrunk commented:
So I should add something to that effect in the data helpers section, you think?

helrond commented:
Sure, that seems reasonable to me...
